### PR TITLE
AG-3258-Chat-Notification-Escalation

### PR DESCRIPTION
**Pull request by sas-wtag**
[AG-3258](https://welltravel.atlassian.net/browse/AG-3258)

### CODE REVIEW

File: proto-libs/agent-platform/userRelationships.proto [hunk 1]
```diff
@@ -11,8 +11,13 @@ message UserRelationsCustomer {
   repeated Model.User travelArrangementClients = 4;
 }
 
+message Empty {
+
+}
```
Comment (luxflux) [inline proto-libs/agent-platform/userRelationships.proto:16]: I think it'd be good to not just respond with an `Empty` response, I think it'd be great to respond with a success/failure:
```suggestion
message SendEmailResponse {
  bool success = 1;
}
```

**Review by luxflux — CHANGES_REQUESTED** [review]

File: proto-libs/agent-platform/userRelationships.proto (hunk 1 above)
Comment (sas-wtag) [reply proto-libs/agent-platform/userRelationships.proto:16]: done

File: proto-libs/chat-backend/chat.proto [hunk 2]
```diff
@@ -108,6 +117,7 @@ service Chat {
   rpc MarkMessageAsSeen(RemoveMessageSeenRequest) returns (RemoveMentionAndSeenResponse) {}
   rpc MarkMentionAsSeen(RemoveMentionRequest) returns (RemoveMentionAndSeenResponse) {}
   rpc GetAllTenants(Empty) returns (stream Tenant) {}
+  rpc MessageSeenStatus(MessageSeenStatusRequest) returns (MessageSeenStatusResponse) {}
```
Comment (luxflux) [inline proto-libs/chat-backend/chat.proto:120]: These changes shouldn't be in here, should they?

File: proto-libs/agent-platform/userRelationships.proto [hunk 3]
```diff
@@ -11,8 +11,23 @@ message UserRelationsCustomer {
   repeated Model.User travelArrangementClients = 4;
 }
 
+message Empty {
```
Comment (luxflux) [inline proto-libs/agent-platform/userRelationships.proto:14]: We don't need this one anymore, do we?

File: proto-libs/agent-platform/userRelationships.proto [hunk 4]
```diff
@@ -11,8 +11,23 @@ message UserRelationsCustomer {
...
+message SendEmailResponse {
+  bool success = 1;
+}
+
 service UserRelationships {
   rpc RetrieveCustomerRelations(Model.User) returns (UserRelationsCustomer) {}
   rpc RetrieveAgentCustomerRelations(Model.User) returns (stream Model.User) {}
   rpc RetrieveAgentColleagueRelations(Model.User) returns (stream Model.User) {}
+  rpc SendEmail(SendEmailRequest) returns (SendEmailResponse) {}
 }
```
Comment (luxflux) [inline proto-libs/agent-platform/userRelationships.proto:33]: I think we should create a new service for this as `SendEmail` isn't really related to `UserRelationships`:
```suggestion
}

service Email {
  rpc Send(SendEmailRequest) returns (SendEmailResponse) {}
}
```

Not quite sure what this means on Agent side, though.

**Review by luxflux — CHANGES_REQUESTED** [review]

File: proto-libs/chat-backend/chat.proto [hunk 5]
```diff
@@ -100,6 +100,15 @@ message Tenant {
   string identifier = 1;
 }
 
+message MessageSeenStatusRequest {
+  int64 messageId = 1;
+  bool isCustomer = 2;
+}
+
+message MessageSeenStatusResponse {
+  string status = 1;
```
Comment (luxflux) [inline proto-libs/chat-backend/chat.proto:109]: Not sure if we should call it `seen` and use a boolean instead?
```suggestion
  bool seen = 1;
```

File: proto-libs/agent-platform/userRelationships.proto (hunk 3 above)
Comment (sas-wtag) [reply proto-libs/agent-platform/userRelationships.proto:14]: It has been removed !

File: proto-libs/chat-backend/chat.proto (hunk 5 above)
Comment (sas-wtag) [reply proto-libs/chat-backend/chat.proto:109]: done

File: proto-libs/agent-platform/userRelationships.proto (hunk 4 above)
Comment (sas-wtag) [reply proto-libs/agent-platform/userRelationships.proto]: added

File: proto-libs/chat-backend/chat.proto [hunk 6]
```diff
@@ -100,6 +100,15 @@ message Tenant {
   string identifier = 1;
 }
 
+message MessageSeenStatusRequest {
+  int64 messageId = 1;
+  bool isCustomer = 2;
```
Comment (luxflux) [inline proto-libs/chat-backend/chat.proto:105]: I think you'd want to use something like this here (related to my other comment in the chat-server):
```suggestion
  int64 messageId = 1;
  Model.User user = 2;
```

File: proto-libs/agent-platform/userRelationships.proto [hunk 7]
```diff
@@ -11,8 +11,22 @@ message UserRelationsCustomer {
...
+
 service UserRelationships {
   rpc RetrieveCustomerRelations(Model.User) returns (UserRelationsCustomer) {}
   rpc RetrieveAgentCustomerRelations(Model.User) returns (stream Model.User) {}
   rpc RetrieveAgentColleagueRelations(Model.User) returns (stream Model.User) {}
 }
+
+service SendEmail {
+  rpc SendEmail(SendEmailRequest) returns (SendEmailResponse) {}
+}
```
Comment (luxflux) [inline proto-libs/agent-platform/userRelationships.proto:32]: This is now somehow included in the user relation ship files. Could you introduce a new `email.proto` file to handle it?

File: proto-libs/agent-platform/userRelationships.proto [hunk 8]
```diff
@@ -11,8 +11,22 @@ message UserRelationsCustomer {
...
+  bool success = 1;
+}
+
 service UserRelationships {
   rpc RetrieveCustomerRelations(Model.User) returns (UserRelationsCustomer) {}
   rpc RetrieveAgentCustomerRelations(Model.User) returns (stream Model.User) {}
   rpc RetrieveAgentColleagueRelations(Model.User) returns (stream Model.User) {}
 }
+
+service SendEmail {
```
Comment (luxflux) [inline proto-libs/agent-platform/userRelationships.proto:30]: Let's name the service less specific so we can add more functionality around email later on:
```suggestion
service Email {
```

**Review by luxflux — CHANGES_REQUESTED** [review]

File: proto-libs/chat-backend/chat.proto (hunk 6 above)
Comment (sas-wtag) [reply proto-libs/chat-backend/chat.proto:105]: done

File: proto-libs/agent-platform/userRelationships.proto (hunk 7 above)
Comment (sas-wtag) [reply proto-libs/agent-platform/userRelationships.proto:32]: done

File: proto-libs/agent-platform/userRelationships.proto (hunk 8 above)
Comment (sas-wtag) [reply proto-libs/agent-platform/userRelationships.proto:30]: done

**Review by luxflux — APPROVED** [review]
Please also update the build commands https://github.com/wtag/blabber/blob/master/package.json#L9-L10

**Review by kup-wtag — APPROVED** [review]

**Review by mal-wtag — APPROVED** [review]

**Pull request merged**